lib/OpenQA/Utils.pm

```diff
@@ -224,6 +224,27 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    my @properties;
+    foreach my $property (@{$needle->{properties}}) {
+        if (ref($property) eq "HASH") {
+            push(@properties, $property);
+            next;
+        }
+        next if ($property ne "workaround");
+
+        my $desc = "";
+        if ($name =~ /\S+\-(bsc\d+|poo\d+)\-\S+/) {
+            $desc = $1;
+        }
+        push(@properties, {name => "workaround", value => $desc});
+        last;
```

Comment:
So all properties after the first occurrence of a string-workaround property are lost. That's likely not what this is supposed to do.

Reply:
I see the problem. I will fix it.